src/TwiggyService.php

```diff
@@ -7,51 +7,33 @@
  * @license GPL-2.0-or-later
  */
 
+declare( strict_types=1 );
+
 namespace Twiggy;
 
-use GlobalVarConfig;
 use Twig\Environment;
 use Twig\Loader\LoaderInterface;
 
 class TwiggyService extends Environment {
-	/**
-	 * Constructor
-	 *
-	 * @param LoaderInterface $loader
-	 * @param array           $mwConfig
-	 * @param array           $options
-	 */
-	public function __construct(LoaderInterface $loader, $mwConfig, $options = []) {
-		parent::__construct($loader, $options);
-		$this->registerPhpFunctionExtension($mwConfig);
-	}
+	public function __construct(
+		LoaderInterface $loader,
+		array $allowedPhpFunctions,
+		array $blacklistedPhpFunctions
+	) {
+		parent::__construct( $loader );
 
-	/**
-	 * Add a Template Location
-	 *
-	 * @param string $namespace
-	 * @param string $directory
-	 *
-	 * @return void
-	 */
-	public function setTemplateLocation(string $namespace, string $directory): void {
-		$this->getLoader()->addPath($directory, $namespace);
+		$functionList = array_filter(
+			array_unique( $allowedPhpFunctions ),
+			fn ( $func ): bool => !in_array( $func, $blacklistedPhpFunctions )
+		);
```

Comment:
`array_diff( $allowedPhpFunctions, $blacklistedPhpFunctions )`

Reply:
I think we can keep `array_unique`, I don't expect it to make a big difference but I haven't had time to test yet what Twig would do with duplicate function definitions using the same name.